UnicornRender/video/include/unicorn/video/Renderer.hpp

```diff
@@ -68,30 +69,31 @@ class Renderer
      */
     UNICORN_EXPORT virtual void SetDepthTest(bool enabled) = 0;
 
-    /** @brief  Creates new geometry mesh
-    *
-    *  Creates and subscribes to mesh.
-    *  Mesh shall be deleted via DeleteMesh().
-    *
-    *  @param [in] material describes mesh visual representation
-    *
-    *  @attention  Mesh lifetime is bound by its renderer's lifetime.
-    *              Using meshes after their renderer was destroyed is undefined behaviour.
-    *              If you're storing mesh pointers, consider storing them with a reference
-    *              to their renderer and listening for its Destroyed event for proper cleanup.
-    *
-    *  @return pointer to newly created geometry::Mesh
-    *
-    *  @sa DeleteMesh
+    /**
+    * @brief Adds mesh to rendering system
+    * @param [in] mesh mesh data
+    * @return true if mesh was successfully added to system
+    */
+    UNICORN_EXPORT virtual bool AddMesh(Mesh* mesh) = 0;
+
+    /**
+    * @brief Adds multiple meshes simultaneously to rendering system
+    * @param [in] meshes meshes data
+    * @return true if meshes was successfully added to system
     */
-    UNICORN_EXPORT virtual Mesh* SpawnMesh(Material const& material) = 0;
+    UNICORN_EXPORT virtual bool AddMeshes(std::list<Mesh*> const& meshes) = 0;
```

Comment:
This method is less useful than a `for` loop with `AddMesh()` - it limits input to a `std::list` and doesn't perform any optimizations for a single call. Also it doesn't provide any feedback to user code on return value, it's not clear what shall be done with a `false`